lib/iris/analysis/_regrid.py

```diff
@@ -390,6 +390,7 @@ def _create_cube(data, src, x_dim, y_dim, src_x_coord, src_y_coord,
         # Create a result cube with the appropriate metadata
         result = iris.cube.Cube(data)
         result.metadata = copy.deepcopy(src.metadata)
+        result.fill_value = src.fill_value
```

Comment:
I could, and it doesn't really matter. IMHO one approach isn't any more correct than the other.

At the time I was identifying all places in the code base where `Cube.metadata` was being assigned to, as now in addition to that, the `fill_value` required to be also set.